apps/db/lib/user/user.ex

```diff
@@ -37,6 +37,32 @@ defmodule DB.User do
     ))
   end
 
+  @doc """
+  Searches for users by `username`. Uses levenshtein distance
+  to determine matches, up to a distance of 5.
+  """
+  def search_users(username: query_string) do
+    DB.Repo.all(
+      from u in User,
+      where: fragment("levenshtein(?, ?)", u.username, ^query_string) < 5,
+      order_by: fragment("levenshtein(?, ?)", u.username, ^query_string),
+      limit: 10
```

Comment:
maybe lets coordinate with kris on what the limit should be. i could see this value as something that is passed in by the client via a query parameter.